libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -724,11 +720,32 @@ void CodeTransform::visitExpression(Expression const& _expression)
 
 void CodeTransform::visitStatements(vector<Statement> const& _statements)
 {
+	// Workaround boost bug:
+	// https://www.boost.org/doc/libs/1_63_0/libs/optional/doc/html/boost_optional/tutorial/gotchas/false_positive_with__wmaybe_uninitialized.html
+	boost::optional<AbstractAssembly::LabelID> jumpTarget = boost::make_optional(false, AbstractAssembly::LabelID());
+
 	for (auto const& statement: _statements)
 	{
 		freeUnusedVariables();
+
```

Comment:
I think it would be better to add a `m_assembly.setSourceLocation(locationOf(statement);` here.

Reply:
Shouldn't that go inside the `if (functionDefinition && !jumpTarget)` block; We only want to set the source location if its the start of a series of function definitions, right?

Reply:
Either you put it before `appendJumpTo` and before `appendLabel` or here - it should be re-set inside the apply_visitor anyway.